src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1342,35 +1343,55 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
-                else if (mode == ::Terminal::SelectionExpansionMode::Cell && !shiftEnabled)
-                {
-                    // Single Click: reset the selection and begin a new one
-                    _terminal->ClearSelection();
-                    _singleClickTouchdownPos = cursorPosition;
-                    _selectionNeedsToBeCopied = false; // there's no selection, so there's nothing to update
-                }
                 else
                 {
-                    // Multi-Click Selection: expand both "start" and "end" selection points
-                    _terminal->MultiClickSelection(terminalPosition, mode);
-                    _selectionNeedsToBeCopied = true;
-                }
+                    // Update the selection appropriately
 
-                _lastMouseClickTimestamp = point.Timestamp();
-                _lastMouseClickPos = cursorPosition;
-                _renderer->TriggerSelection();
+                    // Capture the position of the first click when no selection is active
+                    if (mode == ::Terminal::SelectionExpansionMode::Cell && !_terminal->IsSelectionActive())
+                    {
+                        _singleClickTouchdownPos = cursorPosition;
+                        _lastMouseClickPosNoSelection = cursorPosition;
+                    }
+
+                    // We reset the active selection if one of the conditions apply:
+                    // - shift is not held
+                    // - GH#9384: the position is the same as of the first click starting the selection
+                    // (we need to reset selection on double-click or triple-click, so it captures the word or the line,
+                    // rather than extending the selection)
+                    if (_terminal->IsSelectionActive() && (!shiftEnabled || _lastMouseClickPosNoSelection == cursorPosition))
+                    {
+                        // Reset the selection
+                        _terminal->ClearSelection();
+                        _selectionNeedsToBeCopied = false; // there's no selection, so there's nothing to update
+                    }
+
+                    if (shiftEnabled)
+                    {
+                        if (_terminal->IsSelectionActive())
+                        {
+                            // If there is a selection we extend it using the selection mode
+                            // (expand the "end"selection point)
+                            _terminal->SetSelectionEnd(terminalPosition, mode);
+                        }
+                        else
+                        {
+                            // If there is no selection we establish it using the selected mode
+                            // (expand both "start" and "end" selection points)
+                            _terminal->MultiClickSelection(terminalPosition, mode);
+                        }
+                        _selectionNeedsToBeCopied = true;
+                    }

```

Comment:
IN RETROSPECT: I don't think this is right. We might have lost the case where there's no selection, and you double/triple click on a word

@Don-Vito as an fyi

Reply:
I may stage a revert for this until we can get it in and get it right.

I'd like to see manual test runs for the following scenarios:

* [ ] single click = no selection
* [ ] single click and drag = selection starting from first point
* [ ] single click in unfocused pane and drag = focus pane, selection starting from first point
* [ ] double-click = selects a whole word
* [ ] triple-click = selects a whole line
* [ ] double-click and drag = selects a whole word, drag selects whole words
* [ ] triple-click and drag = selects a whole line, drag selects whole lines
* [ ] <kbd>Shift</kbd> single-click = defines start point
* [ ] second <kbd>Shift</kbd> single-click = defines end point
* [ ] <kbd>Shift</kbd> double-click = selects entire word
* [ ] <kbd>Shift</kbd> triple-click = selects entire line
* [ ] <kbd>Shift</kbd> double-click and drag = selects entire word, drag selects whole words
* [ ] <kbd>Shift</kbd> triple-click and drag = selects entire line, drag selects whole lines